### ISSUE TITLE
OpenVINO Drop in Accuracy

### ISSUE THREAD

**Issue by rhelck** [ultralytics/ultralytics#18345]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I have noticed a large drop in accuracy when exporting to OpenVINO, is this expected and does this match your benchmark results?

Results with OpenVINO
![yolov8_ov](https://github.com/user-attachments/assets/f1c60a33-c529-4848-9b30-dc45032c85b8)

Results with base .pt file
![yolov8_pt](https://github.com/user-attachments/assets/69199bf8-26b9-42f8-97e6-ccbb517a84cb)

### Environment

Ultralytics YOLOv8.1.34 🚀 Python-3.10.12 torch-2.3.1+cu121 CPU (13th Gen Intel Core(TM) i7-13700HX)
Setup complete ✅ (24 CPUs, 30.2 GB RAM, 256.2/1006.9 GB disk)

OS                  Linux-5.15.90.1-microsoft-standard-WSL2+-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.12
Install             pip
RAM                 30.20 GB
CPU                 13th Gen Intel Core(TM) i7-13700HX
CUDA                None

matplotlib          ✅ 3.9.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.9.3>=1.4.1
torch               ✅ 2.3.1>=1.8.0
torchvision         ✅ 0.18.1>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 5.9.8
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0

### Minimal Reproducible Example

```
from ultralytics import YOLO
import cv2 
# # Load a YOLOv8n PyTorch model
model = YOLO("yolov8n.pt")
ov_model = model
# # Export the model
# model.export(format="openvino", half = True)  # creates 'yolov8n_openvino_model/'

# Load the exported OpenVINO model
#ov_model = YOLO("yolov8n_openvino_model/")
img = cv2.imread('bus.jpg')
img = cv2.resize(img, (640,640))
# Run inference
results = ov_model(img, device = 'cpu')
img = results[0].plot()
cv2.imwrite("yolov8_pt.png", img)
for _ in range(5):
    results = ov_model(img, device = 'cpu')
```

### Additional

This may be related to https://github.com/ultralytics/ultralytics/issues/18333

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rhelck, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug your issue further. While it looks like you've shared a reproducible snippet (thank you for that! 🙏), please ensure it includes all the steps leading to the issue (e.g., the export command that creates the OpenVINO model) so we can replicate it precisely.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

🔧 Note that this is an automated response. An Ultralytics engineer will review your issue and provide additional assistance soon if needed!

**Comment by Y-T-G** [comment]
You shouldn't be resizing the image manually

**Comment by Y-T-G** [comment]
Can you try with latest Ultralytics and openvino?

**Comment by rhelck** [comment]
@Y-T-G  I did that in some of these tests for consistency, but point taken
But anyway, when I take out that line I still get these results

**Comment by rhelck** [comment]
@Y-T-G I upgraded to the latest versions of both and I am getting accurate results now, I should have done this immediately.
Thank you!

**Comment by rhelck** [comment]
Kinda surprising that the old version gave such inaccurate results though.

**Comment by glenn-jocher** [comment]
@rhelck older versions may have had compatibility issues or bugs that impacted accuracy during export. We recommend always using the latest versions of both Ultralytics and OpenVINO to ensure optimal performance and compatibility.
